curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md

```diff
@@ -7,57 +7,48 @@ forumTopicId: 302343
 
 # --description--
 
-A vector is defined as having three dimensions as being represented by an ordered collection of three numbers: (X, Y, Z).
+A vector can have one or more values represented by an ordered collection. Examples could be (x), (x, y), or (x, y, z).
 
 # --instructions--
 
-Write a function that takes any numbers of vectors (arrays) as input and computes their dot product. Your function should return `null` on invalid inputs such as vectors of different lengths.
+Write a function that takes two vectors (represented as one-dimensional arrays) as input and computes their dot product. Your function should return `null` on invalid inputs such as vectors of different lengths or passing anything other than two vectors.
 
 # --hints--
 
-dotProduct should be a function.
+`dotProduct` should be a function.
 
 ```js
 assert.equal(typeof dotProduct, 'function');
 ```
 
-dotProduct() should return null.
+`dotProduct()` should return `null`.
 
 ```js
 assert.equal(dotProduct(), null);
 ```
 
-dotProduct(\[[1], [1]]) should return 1.
+`dotProduct(\[[1], [1]])` should return `1`.
 
 ```js
 assert.equal(dotProduct([1], [1]), 1);
 ```
 
-dotProduct(\[[1], [1, 2]]) should return null.
+`dotProduct(\[[1], [1, 2]])` should return `null`.
 
 ```js
 assert.equal(dotProduct([1], [1, 2]), null);
 ```
 
-dotProduct([1, 3, -5], [4, -2, -1]) should return 3.
+`dotProduct([1, 3, -5], [4, -2, -1])` should return `3`.
 
 ```js
 assert.equal(dotProduct([1, 3, -5], [4, -2, -1]), 3);
 ```
 
```

Comment:
````suggestion

`dotProduct([3, 2, 1], [2, 4, 2], [5, 3, 1])` should return `null`.

```js
assert.equal(dotProduct([3, 2, 1], [2, 4, 2], [5, 3, 1]), null);
```

````

Reply:
@ShaunSHamilton Was the change part of the original PR?

Reply:
This specifically, I do not remember so. However, we are not properly testing the user requirement to return `null` on any input other than 2 equally-sized vectors.

Reply:
True.  I will commit your suggestion.